comid-triples.cddl

```diff
@@ -0,0 +1,172 @@
+concise-mid-tag = {
+  ; TODO(tho) language
+  ; TODO(tho) tag-id
+  ; TODO(tho) module-name
+  ; TODO(tho) entity
+  ; TODO(tho) linked-tags
+  comid.triples => triples-map
+  ; TODO(tho) extension
+}
+
+triples-map = non-empty<{
+  ? comid.reference-triples => one-or-more<reference-triple-record>
+  ? comid.endorsed-triples => one-or-more<endorsed-triple-record>
+  ? comid.attest-key-triples => one-or-more<attest-key-triple-record>
+  ? comid.identity-triples => one-or-more<identity-triple-record>
+  * $$triples-map-extension
+}>
+
+; REFERENCE-VALUE triple
+; "target environment $TE" "has reference measurements" "$RV"
+; NOTE(tho): can an attesting evironement have reference values?
+reference-triple-record = [
+  environment-map ; target environment
+  one-or-more<measurement-map> ; reference measurements
+]
+
+; ENDORSED-VALUE triple
+; "environment $E" "has endorsed measurements" "$EV"
+endorsed-triple-record = [
+  environment-map ; (target or attesting) environment
+  one-or-more<measurement-map> ; endorsed measurements
+]
+
+; ATTESTATION-VERIFICATION-KEY triple
+; "attesting environment $AE" "signs Evidence that can be verified with key" "$K"
+attest-key-triple-record = [
+  environment-map ; attesting environment
+  one-or-more<verification-key> ; attestation verification key(s)
+]
+
+; DEVICE-IDENTITY triple
+; "device $D" "is identified by key" "$K"
+identity-triple-record = [
+  environment-map ; device identifier (instance or class)
+  one-or-more<verification-key> ; DevID, or semantically equivalent
+]
+
+verification-key = {
+  ; Verification key (typically a public key)
+  comid.key => COSE_Key
+  ; Optional X.509 certificate chain corresponding to the public key
+  ; in comid.key, encoded as an array of one or more base64-encoded
+  ; DER PKIX certificates.  The certificate containing the public key
+  ; in comid.key MUST be the first certificate.  This MAY be followed
+  ; by additional ; certificates, with each subsequent certificate
+  ; being the one used to certify the previous one.
+  ? comid.keychain => [ + tstr ]
+}
+
+environment-map = non-empty<{
+  ? comid.class => class-map
+  ? comid.instance => $instance-id-type-choice
+}>
```

Comment:
I thought we agreed to extend environment-map  but not the class map?

Reply:
I thought the end result of the conversation was: "we don't want to water down the triple semantics by allowing extensibility of the subject", but maybe I misinterpreted the outcome.  In any case I hold no strong opinions, so I'm happy either way.

Reply:
That was my understanding too. The only other use cases that might use it is groups. Potentially there will be supply chain situations where two Endorsers refer to the same environment using different class-id values.  But this should be a new triple IMO.

Reply:
As per my understanding the water tight implementation was that Class and Instance does not need extensibility so that not to confuse the verifier?
We should certainly keep scope for identifiers which does not fall into Class and Instance scope and let them use the model.
One specific example is "Group" and perhaps their may be many more hidden. Blocking them here does not seem to be the right approach here!

Reply:
I don't think I have a strong opinion. I agree group-id is likely the other one that will be added later. We haven't discussed hardware 'module' as another possibility as we decided not to do 6x triples right now. Possibly we could address 6x semantics by adding a 'module-map' such that if both module-map and class-map are together in the environment-map the semantics are similar to 6x composition? But it is potentially ambiguous as to which contains the other. So I still prefer doing composition as a triple.

Reply:
I am sure we will have this discussion (if we do not add an extension ) when this goes for a review in a wider group!

Reply:
I don't feel strongly about it. If you're concerned about it then its OK with me to add an extension point here.